src/validation.cpp

```diff
@@ -3734,7 +3734,7 @@ bool CChainState::LoadBlockIndex(const Consensus::Params& consensus_params, CBlo
             setDirtyBlockIndex.insert(pindex);
         }
         if (pindex->IsValid(BLOCK_VALID_TRANSACTIONS) && (pindex->HaveTxsDownloaded() || pindex->pprev == nullptr))
-            setBlockIndexCandidates.insert(pindex);
+            ::ChainstateActive().setBlockIndexCandidates.insert(pindex);
```

Comment:
In commit "refactoring: move block metadata structures into BlockManager" (3a8f92e51f60287492d1a1bce5e5e58ae029bb67)

This BlockManager -> ChainstateActive() dependency seems like it will make BlockManager harder to test independently and have confusing side effects. Would suggest reverting this line and adding an explicit `std::set<CBlockIndex*, CBlockIndexWorkComparator>& setBlockIndexCandidates` output parameter to this function. It's only called one place, and I think it'd make the call site clearer, because otherwise you might not know calling this has any effect on the active chain.